### PR TITLE
Extended responseDecodable for JSONDecoder injection

### PR DESCRIPTION
**Pull request by timbms**
And also added test for responseDecodable

### CODE REVIEW

File: Sources/Alamofire+Promise.swift [hunk 1]
```diff
@@ -87,15 +87,21 @@ extension Alamofire.DataRequest {
     }
 
 #if swift(>=3.2)
-    /// Adds a handler to be called once the request has finished.
-    public func responseDecodable<T: Decodable>(queue: DispatchQueue? = nil) -> Promise<T> {
+    /**
+     Returns a Promise for a DecJodable
```
Comment (mxcl) [inline Sources/Alamofire+Promise.swift]: typo

File: Sources/Alamofire+Promise.swift [hunk 2]
```diff
@@ -87,15 +87,21 @@ extension Alamofire.DataRequest {
...
+    public func responseDecodable<T: Decodable>(queue: DispatchQueue? = nil, decoder: JSONDecoder = JSONDecoder()) -> Promise<T> {
         return Promise { seal in
             responseData(queue: queue) { response in
                 switch response.result {
                 case .success(let value):
                     do {
-                        seal.fulfill(try JSONDecoder().decode(T.self, from: value))
-                    } catch {
+                        seal.fulfill(try decoder.decode(T.self, from: value))
+                    } catch let error {
```
Comment (mxcl) [inline Sources/Alamofire+Promise.swift]: this is an unnecessary addition

**Review by mxcl — CHANGES_REQUESTED** [review]
Great! Two minor issues.

**Comment by mxcl** [comment]
Also, tests require a `swift(>=3.2)` guard.

**Comment by timbms** [comment]
Done and committed
It is not clear to me how I can run Travis CI on it.

**Comment by mxcl** [comment]
Sorry somehow I missed your commit.

> It is not clear to me how I can run Travis CI on it.

Travis runs automatically. You can see the results here on the PR.

**Review by mxcl — APPROVED** [review]

**Pull request merged**